source/MaterialXRender/Handlers/stbImageLoader.cpp

```diff
@@ -0,0 +1,98 @@
+#include <MaterialXRender/Window/HardwarePlatform.h>
+
+#if defined(OSWin_)
+#pragma warning( push )
+#pragma warning( disable: 4100)
+#endif
+
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb/stb_image_write.h>
+
+#define STB_IMAGE_IMPLEMENTATION
+#include <stb/stb_image.h>
+
+#include <MaterialXRender/Handlers/stbImageLoader.h>
+
+namespace MaterialX
+{
+bool stbImageLoader::saveImage(const std::string& fileName,
+                                    const ImageDesc& imageDesc)
+{
+    int returnValue = -1;
+
+    int w = static_cast<int>(imageDesc.width);
+    int h = static_cast<int>(imageDesc.height);
+    int channels = static_cast<int>(imageDesc.channelCount);
+    void* data = imageDesc.resourceBuffer;
+
+    std::string extension = (fileName.substr(fileName.find_last_of(".") + 1));
```

Comment:
Not the cleanest interface. Need to call different function based on type. Not
all types which can be loaded can be written as well.